test/test_arhp.rb

```diff
@@ -180,4 +196,16 @@ def assert_action_uses_correct_database(action, sql)
       assert_equal desired_db, current_database(klass)
     end
   end
+
+  def simulate_rails_app_active_record_railties
+    if ActiveRecord.version >= Gem::Version.new('6.0')
+      # Necessary for testing ActiveRecord 6.0 which uses the connection
+      # handlers when clearing query caches across all handlers when
+      # an operation that dirties the cache isinvolved (e.g. create/insert,
+      # update, delete/destroy, truncate, etc.)
+      ActiveRecord::Base.connection_handlers = {
+        ActiveRecord::Base.writing_role => ActiveRecord::Base.default_connection_handler
+      }
```

Comment:
This is to simulate https://github.com/rails/rails/blob/6-0-stable/activerecord/lib/active_record/railtie.rb#L199. This is necessary to create the environment where the underlying database can be switched during a dirtying operation (e.g. insert, update, etc).